tests/unit/ShareStructure.spec.ts

```diff
@@ -0,0 +1,393 @@
+    expect(wrapper.emitted(removeSeriesEvent)[0][0]).toStrictEqual(0)
+    wrapper.destroy()
+  })
+
+  it('Emits cancel event', async () => {
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareClass, -1, 1, null, [])
+    wrapper.find(cancelButtonSelector).trigger('click')
+    expect(wrapper.emitted().resetEvent).toBeTruthy()
+    expect(wrapper.emitted(formResetEvent).length).toBe(1)
+    wrapper.destroy()
+  })
+
+  it('Shows error message for duplicate share class name', async () => {
+    const existingShareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class B', true, 100, true, 0.50, 'CAD', true)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareClass, -1, 1, null, [existingShareClass])
+    const inputElement: Wrapper<Vue> = wrapper.find(nameSelector)
+    inputElement.setValue('Class A')
+    inputElement.trigger('change')
+    await wrapper.vm.$nextTick()
+    await flushPromises()
+    await wrapper.vm.$nextTick()
+    expect(wrapper.find(formSelector).text()).toContain('Class name must be unique')
+    expect(wrapper.vm.$data.formValid).toBe(false)
+    wrapper.destroy()
+  })
+
+  it('Shows error message for duplicate share series name', async () => {
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const shareSeries1 = createShareStructure(1, 1, 'Series', 'Series A', true, 100, true, 0.50, 'CAD', true)
+    const shareSeries2 = createShareStructure(1, 1, 'Series', 'Series B', true, 100, true, 0.50, 'CAD', true)
+    shareClass.series.push(shareSeries1)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareSeries2, -1, 1, 0, [shareClass])
+    const inputElement: Wrapper<Vue> = wrapper.find(nameSelector)
+    inputElement.setValue('Series A')
+    inputElement.trigger('change')
+    await wrapper.vm.$nextTick()
+    await flushPromises()
+    await wrapper.vm.$nextTick()
+    expect(wrapper.find(formSelector).text()).toContain('Series name must be unique')
+    expect(wrapper.vm.$data.formValid).toBe(false)
+    wrapper.destroy()
+  })
+
+  it('Shows error message if class name contains the word value', async () => {
+    const existingShareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class B', true, 100, true, 0.50, 'CAD', true)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareClass, -1, 1, null, [existingShareClass])
+    const inputElement: Wrapper<Vue> = wrapper.find(nameSelector)
+    inputElement.setValue('Class A value')
+    inputElement.trigger('change')
+    await wrapper.vm.$nextTick()
+    await flushPromises()
+    await wrapper.vm.$nextTick()
```

Comment:
I (we) will have to remember that sometimes all these must be awaited for a change event to complete. I know that in other places I (we) just gave up and called the handler method directly, but if the above works then we can go back and fix those other tests.
Thanks for persevering and finding a solution here! 👍 👍

Reply:
I see that this set of 3 awaits is used in a few places...
How about creating a helper method to replace them?

Reply:
Moved it into separate method